### DISCUSSION TITLE
Activity Ranking

### DISCUSSION THREAD

**Discussion by kartikeyaa-k** [exoplatform/exo-community#23] [Q&A]
Hi 
Does exo provide the ranking mechanism for fetching activities?

**Comment by rdenarie** [comment]
Hello,
I'm not sure to understand your question.

Do you mean how activities are sorted when you fetch with API ? 
Which endpoint do you use ? 
Normally it is sorted from the more recent to the older

**Comment by kartikeyaa-k** [comment]
"Do you mean how activities are sorted when you fetch with API ?"
- Yes, I understand they are sorted from the more recent to the older. 

But what if I want to sort they as per various factors, is that possible, for example a post added by a certain user to be showed at the top, is that possible? 


We are using getStream.io (And we would like to switch to Exo), and this is what they have 
https://getstream.io/blog/getting-started-ranked-feeds-getstream-io/

So my question is : Is it possible to do something like on Exo platform?

**Comment by rdenarie** [comment]
Hello,
There is no option to change the order of fetched activities, but there are some option available when fetching activities :

- search activities with time before a certain date
- search activities with time after a certain date
- filter activities on some criteria : 
![image](https://github.com/exoplatform/exo-community/assets/807839/2dd9c79c-360a-4d9a-8e07-2b19b6768399)
  - All
  - Pinned activities
  - My activities
  - My favorites activities
  - Activities in spaces I manage
  - Activities in my favorites spaces

In each case, activities are sorted by time
